src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -251,6 +251,15 @@ case class AppDefinition(
 
   def isUpgrade(to: AppDefinition): Boolean =
     this != to.copy(instances = instances, version = version)
+
+  override def equals(obj: Any): Boolean = {
```

Comment:
you may not do that because that makes the "isUpgrade" stuff and other things fail.

At least, we need to modify some other things for that.